source/bson/extended-json.txt

```diff
@@ -0,0 +1,149 @@
+  RFC <https://www.json.org/json-en.html>`__. This is the format that
+  :dbtools:`mongoexport </mongoexport/>` produces and
+  :dbtools:`mongoimport </mongoimport/>` consumes.
+
+- Shell Mode: a superset of JSON that the :mongosh:`MongoDB shell </>` can parse.
+
+Furthermore, the ``Document`` type provides two sets of convenience
+methods for this purpose:
+
+- ``Document.toJson()``: a set of overloaded methods that convert a
+  ``Document`` instance to a JSON string
+- ``Document(<json>)``: a set of overloaded static factory methods that
+  convert a JSON string to a ``Document`` instance
+
+Writing JSON
+------------
+
+Consider the task of implementing a ``mongoexport``-like tool by using
+the driver:
+
+.. code-block:: scala
+
+   val fileName =    // initialize to the path of the file to write to
+   val collection =  // initialize the collection from which you want to query
+   
+   val writer: PrintWriter = new PrintWriter(fileName)
+   collection.find().subscribe(
+         (doc: Document) => output.write(s"${doc.toJson}\r\n"),
+         (t: Throwable) => // handle failure,
+         () => output.close())
+
+The ``Document.toJson()`` method constructs an instance of a ``JsonWriter`` with
+its default settings, which writes in strict mode with no new lines
+or indentation.
+
+You can override this default behavior by using one of the overloads of
+``toJson()``. As an example, consider the task of writing a JSON string
+that can be copied and pasted into the MongoDB shell:
+
+.. code-block:: scala
+
+   import java.text.SimpleDateFormat
+   
+   val fmt = new SimpleDateFormat("dd/MM/yy")
+   val first = fmt.parse("01/01/2014")
+   val second = fmt.parse("01/01/2015")
+   val doc = Document("startDate" -> Document("$gt" -> first, "$lt" -> second))
+   println(doc.toJson(new JsonWriterSettings(JsonMode.SHELL)))
+
+This code snippet will print out MongoDB shell-compatible JSON, which
+can then be pasted into the shell:
+
+.. code-block:: shell
+
+   { "startDate" : { "$gt" : ISODate("2014-01-01T05:00:00.000Z"), "$lt" : ISODate("2015-01-01T05:00:00.000Z") } }
+
+Reading JSON
+------------
+
+Consider the task of implementing a ``mongoimport``-like tool by using
+driver:
+
+.. code-block:: scala
+
+   import scala.io.Source
+   val fileName =    // initialize to the path of the file to read from
+   val collection =  // initialize the collection from which you want to import to
+   
+   try {
+     for (json <- Source.fromFile(fileName).getLines()) {
+       collection.insertOne(Document(json)).head()
+     }
+   } catch {
+     case ex: Exception => println("Bummer, an exception happened.")
+   }
+
+The ``Document(<json>)`` companion helper method constructs an instance of a
+``JsonReader`` with the given string and returns an instance of an
+equivalent ``Document`` instance. ``JsonReader`` automatically detects the JSON
+flavor in the string, so you do not need to specify it.
```

Comment:
S: looks like there's a note included in the [original document](https://mongodb.github.io/mongo-java-driver/5.0/driver-scala/bson/scala-extended-json/#reading-json) that's missing here, so I'd add that

Reply:
I can't locate the source repo for the `tools` link in that note, so I left it out.

Reply:
Ah gotcha, I agree with leaving it out then!